### ISSUE TITLE
Bug report: ansible-galaxy fails to download git-based roles (2.0.0.2)

### ISSUE THREAD

**Issue by skorochkin** [ansible/ansible#13910]
**Issue Type**: Bug Report
**Ansible Version**:

```
ansible 2.0.0.2
  config file =
  configured module search path = Default w/o overrides
```

**Ansible Configuration**: stock/vanilla (fresly installed from _pip_)
**Environment**: Mac OS X 10.11.2
**Summary**:
When installing custom roles using **ansible-galaxy** from local git repo, it fails to resolve provided url scheme (scm+url,branch). Also tried following: **http**, **git+https** and **git**, but no any of them have worked.
**Steps To Reproduce**:

```
ansible-galaxy install -p . git+http://gitlab/ansible/somerole.git,master
```

**Expected Results**: (from v1.9.3)

```
- executing: git clone http://gitlab/ansible/somerole.git somerole
- executing: git archive --prefix=somerole/ --output=/var/folders/n_/6trpndtn74b_595vqxg_29080000gn/T/tmp5BIBPo.tar master
- extracting somerole to ./somerole
- somerole was installed successfully
```

**Actual Results**:

```
- downloading role from git+http://gitlab/ansible/somerole.git,master
 [ERROR]: failed to download the file: <urlopen error unknown url type: git+http>

 [WARNING]: - git+http://gitlab/ansible/somerole.git,master was NOT installed successfully.

ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
```

**Comment by bborysenko** [comment]
Just to confirm that issue is really present with 2.0.0.2

**Comment by bcoca** [comment]
i can install git roles w/o issues, I need more information to reproduce this

**Comment by skorochkin** [comment]
@bcoca well, if you compare **stable_1.9** `execute_install` (`lib/ansible/galaxy/role.py`) method and **devel** (**2.1**) `execute_install` (`lib/ansible/cli/galaxy.py`) + `install` (`lib/ansible/galaxy/role.py`) methods , you''ll notice that: **1.9** version has method `scm_archive_role` which is used for scm based galaxy installs, while in **devel** there is only one method `fetch` which is based only on `open_url` (that's where `<urlopen error unknown url type: git+http>` exception is coming from)

**Comment by skorochkin** [comment]
@bcoca well, that is the `ansible.galaxy.role.GalaxyRole` object passed (in my case) to `role.install()` from `galaxy.execute_install()`:

```
{
'_install_info': None,
 '_metadata': None,
 'galaxy': <ansible.galaxy.Galaxy object at 0x10b073c10>,
 'name': 'git+http://gitlab/ansible/somerole,master',
 'options': <Values at 0x10afc20e0: {'ignore_certs': False, 'api_server': 'https://galaxy.ansible.com', 'no_deps': False, 'verbosity': 4, 'role_file': None, 'roles_path': '/etc/ansible/roles', 'ignore_errors': False, 'force': False}>,
 'path': '/etc/ansible/roles/git+http://gitlab/ansible/somerole,master',
 'scm': None,
 'src': 'git+http://gitlab/ansible/somerole,master',
 'version': None
}
```

as you can see, scm is `None`, so `role.install()` treats it as usual url instead of scm based url and following `if` statement never get executed:

```
...
if self.scm:
            # create tar file from scm url
            tmp_file = RoleRequirement.scm_archive_role(**self.spec)
...
```

From my understanding we're missing `role_spec_parse` somewhere before constructing role.

**Comment by bcoca** [comment]
I realize now that all our tests use requirements files, which means the 'inline url' is not covered. We might just need to add detection to it to properly set scm.

**Comment by bcoca** [comment]
closing via merged #14044
